tfhe/src/zk.rs

```diff
@@ -24,6 +34,65 @@ impl Named for CompactPkePublicParams {
     const NAME: &'static str = "zk::CompactPkePublicParams";
 }
 
+pub struct CompactPkePublicParamsConformanceParams {
+    lwe_dim: LweDimension,
+    max_num_message: usize,
+    noise_bound: u64,
+    ciphertext_modulus: u64,
+    plaintext_modulus: u64,
+    msbs_zero_padding_bit_count: ZkMSBZeroPaddingBitCount,
+}
+
+impl CompactPkePublicParamsConformanceParams {
+    pub fn new<E, P: TryInto<CompactPublicKeyEncryptionParameters, Error = E>>(
+        value: P,
+        max_num_message: usize,
+    ) -> Result<Self, crate::Error>
+    where
+        E: Into<crate::Error>,
+    {
+        let params: CompactPublicKeyEncryptionParameters =
+            value.try_into().map_err(|e| e.into())?;
+
+        let mut plaintext_modulus = (params.message_modulus.0 * params.carry_modulus.0) as u64;
+        // Add 1 bit of modulus for the padding bit
+        plaintext_modulus *= 2;
+
+        let (lwe_dim, max_num_message, noise_bound, ciphertext_modulus, plaintext_modulus) =
+            CompactPkeCrs::prepare_crs_parameters(
+                params.encryption_lwe_dimension,
+                max_num_message,
+                params.encryption_noise_distribution,
+                params.ciphertext_modulus,
+                plaintext_modulus,
+            )?;
+
+        Ok(Self {
+            lwe_dim,
+            max_num_message,
+            noise_bound,
+            ciphertext_modulus,
+            plaintext_modulus,
+            // CRS created from shortint params have 1 MSB 0bit
+            msbs_zero_padding_bit_count: ZkMSBZeroPaddingBitCount(1),
+        })
+    }
+}
+
+impl ParameterSetConformant for CompactPkePublicParams {
```

Comment:
Not sure to understand the need to check conformance on public parameters

It looks to me that if we can trust a `CompactPkePublicParamsConformanceParams`, we can also trust a ` CompactPkePublicParams`

Reply:
The `CompactPkePublicParamsConformanceParam` can be derived from the public key encryption parameters. 

Concretely if you receive a key and a crs, this checks that they are compatible.